chapter7.adoc

```diff
@@ -25,77 +25,157 @@ to the RDSM. The RDSM, in the course of handling an external interrupt, should
+    bits 28:24     CIXS (WARL)
+    bits 22:20     LHXS (WARL)
+    bits 11:0      High Base PPN (WARL)
 ....
 
-Although a variety of interrupt controllers can be activated for external
-interrupts at the S- and VS-levels, a singular interrupt controller manages
-external interrupts at the M-level and is perpetually active. The `Smsdia`
-extension does not impact external interrupts at the M-level.
-
-When an interrupt controller has an S-level external interrupt awaiting
-resolution at the S-level on a hart, an S-level-external-interrupt-pending
-signal is asserted. Certain interrupt controllers, like the IMSIC, support
-delivering external interrupts directly to the VS-level. An IMSIC, for instance,
-supports up to 63 guest interrupt register files, each potentially linked to a
-virtual machine that the hypervisor might schedule for execution on a hart. Each
-guest interrupt register file can assert a VS-level-external-interrupt-pending
-signal when a VS-level external interrupt is pending resolution by a guest
-operating at VS-level. With the introduction of supervisor domains, a multitude
-of S-level and VS-level external interrupt pending signals might be asserted by
-the various interrupt controllers connected to a hart.
-
-Interrupt controllers are configured through a memory-mapped register interface,
-a collection of CSRs, or a blend of both. For instance, software interfaces with
-an APLIC via a memory-mapped register interface. In another case, software
-interacts with an IMSIC through a set of S and VS-level CSRs, along with a
-series of memory-mapped registers.
-
-The `Smsdia` supports the connection of multiple interrupt controllers to a
-hart, allowing a supervisor domain to be paired with one of these interrupt
-controllers. The `Smsdia` extension uses the `msdcfg` CSR to signify the
-active interrupt controller for a supervisor domain. When CSRs are employed to
-interface with the interrupt controller state, they interact with the state
-corresponding to the active interrupt controller.
-
-The supervisor domain interrupt controllers connected to a hart may not be
-identical. For example, each supervisor domain IMSIC may support a different
-number of external interrupt identities for S- and/or VS-level, and the number
-of guest interrupt files supported by each supervisor domain IMSIC may not be
-identical.
-
-Upon scheduling a supervisor domain for execution on a hart, the RDSM requires a
-mechanism to appoint the linked interrupt controller as the active one for the
-hart. Selecting an interrupt controller also chooses the linked S and VS-level
-interrupt pending signals as those detected by the hart for instigating the
-associated traps. When CSRs are utilized to interface with an interrupt
-controller's state, the interaction is with the active interrupt controller.
+If `CI` is the Child Index from the `source[i]` corresponding to an interrupt
+source `_i_` then the MSI target address is thus computed as follows:
```

Comment:
Hi @ved-rivos 
The name of register is `sourcecfg[i]` not `source[i]`.

IIUC, the `sourcecfg[i]` here means the `sourcecfg[i]` in parent machine-level interrupt domain, instead of the `sourcecfg[i]` in child supervisor-level interrupt domain.
Would it be better to explicitly state this here for better readability?

Reply:
> If `CI` is the Child Index from the ...

Besides, why using the word "If" here ?
Using "If" means there are other interpretations of `CI`.
Does the spec intend to do this ?

Reply:
> IIUC, the sourcecfg[i] here means the sourcecfg[i] in parent machine-level interrupt domain, instead of the sourcecfg[i] in child supervisor-level interrupt domain.

The parent of any supervisor-level interrupt domain is always a machine-level domain.

Reply:
> Besides, why using the word "If" here ?

The word was used to establish the variable `CI`. Will reword it if it is confusing.